ui/components/ConfirmationModal.js

```diff
@@ -383,10 +385,14 @@ function ConfirmationMsg(props) {
                     )}
                   </div>
                 }
+                disabled={!CAN(keys.VALIDATE_DESIGN.action, keys.VALIDATE_DESIGN.resource)}
               />
             )}
             <Tab
-              disabled={disabled}
+              disabled={
+                !CAN(keys.UNDEPLOY_DESIGN.action, keys.UNDEPLOY_DESIGN.subject) ||
+                (CAN(keys.UNDEPLOY_DESIGN.action, keys.UNDEPLOY_DESIGN.subject) && !disabled)
+              }
```

Comment:
We can simplify this, any reason we should stick with the change you made?

```suggestion
              disabled={!CAN(keys.UNDEPLOY_DESIGN.action, keys.UNDEPLOY_DESIGN.subject) || !disabled}
```

Reply:
@Yashsharma1911 I wrote it that way because we need to first check if the person has the permission to do that action or not, and then check if the disabled property (which is dependent on the length of the k8s context.) is true or false.
Your suggestion has the chance to enable the action even if the person doesn't have permission for that action.

Reply:
Though I just realised the '!' mark before diabled isn't needed and have updated the PR for the same.